database_size/models/ir_model_size.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2025 Opener B.V. <https://opener.amsterdam>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from odoo import api, fields, models
+
+
+class IrModelSize(models.Model):
+    _name = "ir.model.size"
+    _description = "Disk space usage per model"
+    _order = "measurement_date desc, total_model_size desc"
+    _rec_name = "model_id"
+    _sql_constraints = [
+        (
+            "uniq_model_id_measurement_date",
+            "unique(model_id, measurement_date)",
+            "There is already a measurement for this model on the given date",
+        ),
+    ]
+    model_id = fields.Many2one(
+        comodel_name="ir.model",
+        index=True,
+        ondelete="cascade",
+        required=True,
+    )
+    measurement_date = fields.Date(
+        "Date of Measurement",
+        help="For the exact time, check the record's write_date.",
+        required=True,
+    )
+    total_model_size = fields.Integer(
+        compute="_compute_total_sizes",
+        help="Total model size in MB. This includes attachments.",
+        store=True,
+    )
+    total_database_size = fields.Integer(
+        compute="_compute_total_sizes",
+        help="Total Model Size in MB. This includes many2many tables",
+        store=True,
+    )
+    total_table_size = fields.Integer(
+        help="Total Table Size in MB. This includes indexes and toast tables",
+    )
+    table_size = fields.Integer(
+        string="Bare Table Size",
+        help="Bare Table Size in MB.",
+    )
```

Comment:
```suggestion
    table_size = fields.Integer(
        string="Bare Table Size",
        help="Bare Table Size, in Bytes",
    )

    table_size_mb = fields.Float(
        compute=lambda x: x._compute_mb_size("table_size"),
        string="Bare Table Size (MB)",
        help="Bare Table Size in MB.",
    )
    def _compute_mb_size(self, field_name):
        for item in self:
            setattr(item, field_name + "_mb", getattr(item, field_name))
```

With the current design, information under MB are lost. I understand that In most cases, information rounded to the MB is sufficient, but it's a shame to lose a level of information. 
Also, a table can be assumed to be empty if its size is < 1MB.

I therefore suggest keeping the value in Bytes (Integer) , and having a calculated field (float) in MB.

Reply:
The aim of the module is to check for large tables and their growth. Many2many relations are hardly ever symmetric, far from it. Therefore, if the model related table grows, the many2many table grows, and trimming down the large model table will also trim down the many2many table. It would be misleading to attribute the many2many table on any other criteria than size. Indeed, if you start out with 40 partner categories and 10 partners, the linked model will switch but this will only be the case when the table sizes are trivially small. So I would argue against your suggestion.

Reply:
The aim of this module is to monitor the size per model, especially the growth of that size.  Keeping track of very small tables, and of very small differences only distract from that aim. It will create additional ir.model.size records that show up when inspecting the report (Unrounded numbers were my first implementation, but I decided to go for MB when I first saw the huge lists that it produced, covering lots of unused or marginal tables). When comparing two dates, the records with only minimal changes will clutter up the screen when trying to find out which tables changed size in a meaningful way. By only taking megabytes into account, the usability of the module improves and the data it produces remains manageable. I can't think of a benefit to keep track of smaller quantities at the moment. Do you have a concrete goal that can be achieved by this?

Reply:
Hi.

> It will create additional ir.model.size records that show up when inspecting the report (

Well, I used the current implementation, and it looks that there are record for "very small tables", so I'm not sure to understand.

![image](https://github.com/user-attachments/assets/37678944-6bdd-4fae-a9eb-714368b59431)

> Do you have a concrete goal that can be achieved by this?

No. It was a reflex to think in terms of : “Being precise increases potential uses.”

let's go merge the current implementation. If people needs more precise data, it can be changed later quite easely.

Reply:
Thanks for accepting my preference! And you are right, it does not make a difference in the number of records.